### PR TITLE
Feature 4177/persistversioncql

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-4177](https://jira.cms.gov/browse/MAT-4177)
(Optional) Related Tickets:

### Summary
added validations for checking if the CQL is empty, if it has errors and storing the data in Hapi Fhir server 

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included and Code coverage is verified.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java [hunk 1]
```diff
@@ -57,9 +64,52 @@ public CqlLibrary createVersion(String id, boolean isMajor, String username) {
...
+          "User [{}] cannot create a version for CQL Library with id [{}] "
+              + "as the Cql has errors in it",
+          username,
+          cqlLibrary.getId());
+
+      throw new ResourceCannotBeVersionedException(
+          "CQL Library", cqlLibrary.getId(), username, "the Cql has errors in it");
+    }
+
+    if (cqlLibrary.getCql().length() == 0) {
```
Comment (jkotanchik-SB) [inline src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java:99]: Can `getCql()` return null?

Comment (sb-prateekkeerthi) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java:99]: As per the model, CQL can be null and if CQL is empty, getCql returns empty string

**Review by jkotanchik-SB — APPROVED** [review]

File: src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java (hunk 1 above)
Comment (jkotanchik-SB) [reply src/main/java/gov/cms/madie/cqllibraryservice/services/VersionService.java:99]: Sounds good, thanks!
